packages/aws-lambda/src/ssm.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ * (c) Copyright Instana Inc. and contributors 2022
+ */
+
+'use strict';
+
+const ENV_NAME = 'INSTANA_SSM_PARAM_NAME';
+let fetchedValue;
+let envValue;
+let errorFromAWS;
+let initTimeoutInMs = 0;
+
+module.exports.reset = () => {
+  fetchedValue = null;
+  envValue = null;
+  errorFromAWS = null;
+  initTimeoutInMs = 0;
+};
+
+module.exports.isUsed = () => !!envValue;
+
+module.exports.validate = () => {
+  // NOTE: Always reset the env value, we do not wanne cache, because it might change
+  envValue = null;
+
+  const _envValue = process.env[ENV_NAME];
+
+  if (!_envValue || !_envValue.length) {
+    return false;
+  }
+
+  envValue = _envValue;
+
+  return envValue;
+};
+
+module.exports.init = ({ logger }) => {
+  let AWS;
+
+  // CASE: Customer did not set INSTANA_SSM_PARAM_NAME, skip
+  if (!exports.isUsed()) {
+    return;
+  }
+
+  // CASE: We already fetched the SSM value, skip
+  if (fetchedValue) {
+    return;
+  }
+
+  initTimeoutInMs = Date.now();
+
+  try {
+    /**
+     * From https://docs.aws.amazon.com/lambda/latest/dg/lambda-nodejs.html:
+     *
+     * Your code runs in an environment that includes the AWS SDK for JavaScript,
+     * with credentials from an AWS Identity and Access Management (IAM) role that you manage.
+     */
+    // eslint-disable-next-line
+    AWS = require('aws-sdk');
+
+    // https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html
+    AWS.config.update({ region: process.env.AWS_REGION });
```

Comment:
This var is a default var in AWS lambda. Customer does not need to do anything :) 
See https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html

> Lambda runtimes set several environment variables during initialization.

Reply:
I think we should rather pass the region into the AWS.SSM() constructor instead of making this change globally. 

I think new `AWS.SSM({region: process.env.AWS_REGION});` should work as well and it might be less invasive.

See the section "Constructor Details" on https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SSM.html

Reply:
Yeah great idea 👍